src/common/snippets/include/snippets/op/memory_access.hpp

```diff
@@ -48,14 +23,46 @@ class MemoryAccess : public ngraph::op::Op {
 public:
     OPENVINO_OP("MemoryAccess", "SnippetsOpset");
 
+    /**
+    * @interface PortDescriptor
+    * @brief This class describes port of MemoryAccess operation
+    * @param m_count - count of elements to load/store
+    * @param m_offset - starting index of elements to load/store
+    * @param m_index - port index
+    * @ingroup snippets
+    */
+    struct PortDescriptor {
+        PortDescriptor(size_t count, size_t offset) : m_count(count), m_offset(offset) {}
+        PortDescriptor() = default;
+
+        size_t m_count = 0lu;
+        size_t m_offset = 0lu;
+        size_t m_index = 0lu;
+
+    private:
+        PortDescriptor(size_t count, size_t offset, size_t index) : m_count(count), m_offset(offset), m_index(index) {}
+
+        friend class MemoryAccess;
+    };
+
     void set_input_port_descriptor(const PortDescriptor& desc, const size_t i);
     void set_output_port_descriptor(const PortDescriptor& desc, const size_t i);
-    PortDescriptor get_input_port_descriptor(const size_t i) const;
-    PortDescriptor get_output_port_descriptor(const size_t i) const;
-    PortDescriptor& get_input_port_descriptor(const size_t i);
-    PortDescriptor& get_output_port_descriptor(const size_t i);
+    const PortDescriptor& get_input_port_descriptor(const size_t i) const;
+    const PortDescriptor& get_output_port_descriptor(const size_t i) const;
+
+    void set_input_count(size_t count, size_t idx);
+    void set_output_count(size_t count, size_t idx);
```

Comment:
1) It makes sense to have a default `idx = 0` value
2) We used to have only one count for both inputs and outputs, so I'm a little confused here now. If I want to set count for `Load` should I use `set_input_count` or `set_output_count`? And if only `set_input_count` is legal for such operations, what happens if I `set_output_count`? I've seen that you created `set_count` for `Load` and `Store`, and I think this is a right direction, but still `set_input_count` [ could be called](https://github.com/openvinotoolkit/openvino/blob/5deb3c326e8845fd43cce555d70aa066c2f8db9d/src/common/snippets/src/pass/vector_to_scalar.cpp#L27)  on load, which is a bit confusing. I don't insist, but should we make a separate class for single-port operations maybe?

Reply:
1. Nice idea. let's do it, thanks
2. `Load` works with data on input and with vector register on output. So you should work with `set_input_count`. And opposite for `Store`. I thought that it seems like logical things. We can discuss it